dialect/sql/sqlgraph/errors.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019-present Facebook Inc. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package sqlgraph
+
+import (
+	"strings"
+)
+
+// IsConstraintError returns true if the error resulted from a DB constraint error
+func IsConstraintError(err error) bool {
+	return IsUniquenessConstraintError(err) || IsForeignKeyConstraintError(err)
+}
+
+// IsUniquenessConstraintError returns true if the error resulted from a DB uniqueness constraint error
+func IsUniquenessConstraintError(err error) bool {
+	uniquenessErrors := map[string]string{
+		"mysql":    "Error 1062",
+		"postgres": "violates unique constraint",
+		"sqlite":   "UNIQUE constraint failed",
```

Comment:
Use `dialect.MySQL`, `dialect.PostgreSQL`, etc instead of these strings. Why do we need this, and not just `[3]string{...}`?

Reply:
we don't, i'm removing